src/chain.h

```diff
@@ -596,65 +601,100 @@ class CDiskBlockIndex : public CBlockIndex
 
 /** An in-memory indexed chain of blocks. */
 class CChain {
-private:
+protected:
     std::vector<CBlockIndex*> vChain;
     CBlockIndex *lastTip;
-
+    CBlockIndex *at(int nHeight) const REQUIRES(cs_main)
+    {
+        if (nHeight < 0 || nHeight >= (int)vChain.size())
+            return NULL;
+        return vChain[nHeight];
+    }
 public:
     /** Returns the index entry for the genesis block of this chain, or NULL if none. */
-    CBlockIndex *Genesis() const {
+    virtual CBlockIndex *Genesis() const REQUIRES(cs_main) {
         return vChain.size() > 0 ? vChain[0] : NULL;
     }
 
     /** Returns the index entry for the tip of this chain, or NULL if none. */
-    CBlockIndex *Tip() const {
+    virtual CBlockIndex *Tip() const REQUIRES(cs_main) {
         return vChain.size() > 0 ? vChain[vChain.size() - 1] : NULL;
     }
     
     /** Returns the last tip of the chain, or NULL if none. */
-    CBlockIndex *LastTip() const {
+    virtual CBlockIndex *LastTip() const REQUIRES(cs_main) {
```

Comment:
Weren't we going to get rid of the LastTip()? (https://github.com/KomodoPlatform/komodo/pull/486#issuecomment-922465502)

Reply:
Removed LastTip()